### ISSUE TITLE
ical-upcoming is missing (node-red v2.1.4)

### ISSUE THREAD

**Issue by geieraffe** [naimo84/node-red-contrib-ical-events#112]
I just updated my node red to 2.1.4,
and after this my flows are stopped with the message ical-upcoming would be missing.
In the paletts it shows:

"node-red-contrib-ical-events 1.5.3
Error: Cannot find module 'kalender-events' (line:6)
Error: Cannot find module 'kalender-events' (line:4)
Error: Cannot find module 'kalender-events' (line:4)"

Node-RED v2.1.4
Node.js v10.24.1

I'm not sure what I should look for.

**Comment by naimo84** [comment]
Hi @geieraffe,

sorry for the inconveniences... I didn't check if my node is still compatible to node v10 for some time, as the EOL is half a year ago :( https://en.wikipedia.org/wiki/Node.js#Releases

But, I have found the issue, why it can't be loaded by node-red https://github.com/naimo84/kalender-events/commit/ace78e3ebe29cb8c2663ec74229c1e549fc6d0b8#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R36

node v10 can't handle the exports: dist/index.js .. it requires a main: dist/indes.js

TL;DR

You have 3 options:

- upgrade node to minimum v12, if that's possible for you
- npm install kalender-events@0.20.3 in your node-red folder, where node-red-contrib-ical-events is installed
- wait some time for the next version of node-red-contrib-ical-events. But this may take a fews days ;) I'll do my best.

Best regards,
Benjamin

**Comment by geieraffe** [comment]
Thanks @naimo84 ,

at some point I suspected myself that it must be the node version. 
Since also the new node-red version does not support older node versions any more.

I have now uptated to node v14 and it works.

The problem only appeared because I could install the new version of node-red, although it then attracted an old node version. 
I still couldn't figure out why:

```
Already have nodejs v14.17.6
OLD nodejs 14 :
NEW nodejs  :

***************************************

Started : Mo 13. Dez 15:35:13 CET 2021
Running for user pi at /home/pi
Found global nodes:   :

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Paketlisten werden gelesen...
Abhängigkeitsbaum wird aufgebaut....
Statusinformationen werden eingelesen....
Paket »nodered« ist nicht installiert, wird also auch nicht entfernt.
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.

Versions: node:v10.24.1 npm:7.23.0
Now install Node-RED
npm WARN EBADENGINE Unsupported engine { package: 'node-red@2.1.4',
npm WARN EBADENGINE   required: { node: '>=12' },
npm WARN EBADENGINE   current: { node: 'v10.24.1', npm: '7.23.0' } }
npm WARN EBADENGINE Unsupported engine { package: 'fs-extra@10.0.0',
npm WARN EBADENGINE   required: { node: '>=12' },
npm WARN EBADENGINE   current: { node: 'v10.24.1', npm: '7.23.0' } }
npm WARN EBADENGINE Unsupported engine { package: 'node-red-admin@2.2.1',
npm WARN EBADENGINE   required: { node: '>=12' },
npm WARN EBADENGINE   current: { node: 'v10.24.1', npm: '7.23.0' } }
````

You see it first said I already have installed v14 just to say my current version would be v10.24.1.
